ggml/src/CMakeLists.txt

```diff
@@ -561,7 +563,35 @@ if (GGML_VULKAN)
             add_compile_definitions(GGML_VULKAN_RUN_TESTS)
         endif()
 
-        set(GGML_EXTRA_LIBS ${GGML_EXTRA_LIBS} Vulkan::Vulkan)
+        set (_llama_vk_genshaders_cmd ${Python_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/../ggml_vk_generate_shaders.py)
+        set (_llama_vk_header     ${CMAKE_CURRENT_BINARY_DIR}/ggml-vulkan-shaders.hpp)
+        set (_llama_vk_source     ${CMAKE_CURRENT_BINARY_DIR}/ggml-vulkan-shaders.cpp)
+        set (_llama_vk_input_dir  ${CMAKE_CURRENT_SOURCE_DIR}/vulkan-shaders)
+        set (_llama_vk_output_dir ${CMAKE_CURRENT_BINARY_DIR}/vulkan-shaders.spv)
```

Comment:
Not sure how important is this, but the existing `GGML_METAL_EMBED_LIBRARY` option uses `${CMAKE_BINARY_DIR}/autogenerated` instead for the generated files. It might be better to unify this.

Reply:
On the other hand, now that I'm thinking about it technically this is part of the GGML library and so the generated files should maybe be put into `build/ggml/src`.